include/rdma/fi_log.h

```diff
@@ -35,8 +35,6 @@
 #ifndef FI_LOG_H
 #define FI_LOG_H
 
-#include "config.h"
```

Comment:
Note that `fi_log.h` is not a user-visible file -- it's for internal use inside libfabric only (perhaps it should not be installed by `make install`...?).

Hence, it is totally fine to it to use the private `config.h` that is generated by the GNU Autotools.  There will never be the possibility of an application *using* libfabric conflicting with our `config.h` because it's private and not visible to libfabric-using applications.

We do not support any other build system on Linux/OS X -- any road that tries to support multiple systems simultaneously (cmake, autotools, scons, ...etc.) when you don't have to leads to madness.  Supporting Visual Studio in addition to GNU Autotools is going to be bad enough (personally, I'm of the opinion that the GNU Autotools build system is the "master" build system, and the Visual Studio support will need to keep up with whatever changes are made in the GNU Autotools build system -- such as providing a dummy `config.h` file), but there's currently not much of an option if native Windows support is desired.

Adding support for a 3rd build system would be... undesirable.